src/libcollections/str.rs

```diff
@@ -1421,13 +1691,19 @@ impl str {
         result
     }
 
-    /// Returns the lowercase equivalent of this string.
+    /// Returns the lowercase equivalent of this string slice, as a new `String`.
+    ///
+    /// 'Lowercase' is defined according to the terms of the Unicode Derived Core Property
```

Comment:
Might be good to expand more on how the function behaves on more-complex unicode input. E.g. whether the function does case-mapping etc.